sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/implementation/ObjectMapperFactory.java

```diff
@@ -12,44 +14,71 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.SerializationFeature;
 import com.fasterxml.jackson.databind.cfg.MapperBuilder;
+import com.fasterxml.jackson.databind.cfg.PackageVersion;
 import com.fasterxml.jackson.databind.json.JsonMapper;
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 
 /**
  * Constructs and configures {@link ObjectMapper} instances.
  */
 final class ObjectMapperFactory {
+    private static final ClientLogger LOGGER = new ClientLogger(ObjectMapperFactory.class);
+    final boolean useJackson215;
+    private boolean jackson215IsSafe = true;
+
+    ObjectMapperFactory() {
+        this.useJackson215 = PackageVersion.VERSION.getMinorVersion() >= 15
```

Comment:
nit: do we need a sanity check that major version is 2?

Reply:
For now, I think it's safe just to use the minor version, but in the future we'll need to check the major version once Jackson 3 is released.